pkg/entities/record.go

```diff
@@ -121,6 +116,27 @@ func (d *dataRecord) PrepareRecord() error {
 	return nil
 }
 
+func (d *dataRecord) GetBuffer() []byte {
+	if len(d.buffer) == d.len || d.isDecoding {
+		return d.buffer
+	}
+	d.buffer = make([]byte, d.len)
+	index := 0
+	for _, element := range d.orderedElementList {
+		len := GetInfoElementLen(element)
```

Comment:
can this length be store in the element first time instead of calling `GetInfoElementLen` again?

Reply:
Yes. Updated.